Schweizerisches Bundesgericht.js

```diff
@@ -0,0 +1,243 @@
+{
+	"translatorID": "9e898223-938c-4c8f-8b5c-d118b6e44aae",
+	"label": "Schweizerisches Bundesgericht",
+	"creator": "Hans-Peter Oeri",
+	"target": "^https://www\\.bger\\.ch/ext/eurospider/live/(de|fr|it)/php/(clir|aza)/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-11-13 21:17:33"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2020 Hans-Peter Oeri
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+const abbrevPublished = {
+	de: "BGE ",
+	fr: "ATF ",
+	it: "ATF "
+};
+
+const abbrevUnpublished = {
+	de: "BGer ",
+	fr: "TF ",
+	it: "TF "
+};
+
+const courtName = {
+	de: "Schweizerisches Bundesgericht",
+	fr: "Tribunal Fédéral Suisse",
+	it: "Tribunale Federale Svizzero"
+}
+
+/**
+ * read query parameters from url
+ *
+ * @param   {String} url
+ * @returns {{}}
+ */
+function parseURLParameters(url) {
+	let parts = require('url').parse(url, true);
```

Comment:
We don't want to use `require` here, or anywhere in a translator - I am genuinely surprised that it's available in the sandbox at all! Try [`new URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL).

Reply:
Unlike `require` the global URL API is _not_ available in my Scaffold sandbox :( (v5.0.96.3)

That's why I searched `Zotero.Utilities` and found its `parseURL` that itself relies on `require('url')` but does not offer the second parameter (to parse the querystring). If `require` is prohibited and `URL` not supported, is there any other documented way to parse the query string - short of low-level parsing it oneself (which would be a shame of duplicated effort).

Reply:
Could you try [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)? That should be available in the sandbox. (It is in my client.)

Reply:
URLSearchParams works - one probably could include the 'full' URL in a future version of the sandbox ;)